### PR TITLE
Export string sets in monitoring infos.

### PR DESCRIPTION
**Pull request by robertwb**
This ensures they'll be available on Dataflow Runner v2.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by robertwb** [comment]
R: @Abacn

**Comment by Abacn** [comment]
Checkout this PR, removed excludeCategory in commonRunnerV2ExcludeCategories and ran `./gradlew :runners:google-cloud-dataflow-java:validatesRunnerV2Test --tests *test*StringSetMetrics` manually, still get the same AssertionError

```
org.apache.beam.sdk.metrics.MetricsTest$AttemptedMetricTests

java.lang.AssertionError: 
Expected: iterable with items [MetricResult{inNamespace="org.apache.beam.sdk.metrics.MetricsTest", name="sources", step="MyStep1", attempted=<StringSetResult{stringSet=[gcs]}>}, MetricResult{inNamespace="org.apache.beam.sdk.metrics.MetricsTest", name="sinks", step="MyStep2", attempted=<StringSetResult{stringSet=[kafka, bq]}>}, MetricResult{inNamespace="org.apache.beam.sdk.metrics.MetricsTest", name="sideinputs", step="MyStep1", attempted=<StringSetResult{stringSet=[bigtable, spanner]}>}, MetricResult{inNamespace="org.apache.beam.sdk.metrics.MetricsTest", name="sideinputs", step="MyStep2", attempted=<StringSetResult{stringSet=[sql, bigtable]}>}] in any order
     but: no item matches: MetricResult{inNamespace="org.apache.beam.sdk.metrics.MetricsTest", name="sources", step="MyStep1", attempted=<StringSetResult{stringSet=[gcs]}>}, MetricResult{inNamespace="org.apache.beam.sdk.metrics.MetricsTest", name="sinks", step="MyStep2", attempted=<StringSetResult{stringSet=[kafka, bq]}>}, MetricResult{inNamespace="org.apache.beam.sdk.metrics.MetricsTest", name="sideinputs", step="MyStep1", attempted=<StringSetResult{stringSet=[bigtable, spanner]}>}, MetricResult{inNamespace="org.apache.beam.sdk.metrics.MetricsTest", name="sideinputs", step="MyStep2", attempted=<StringSetResult{stringSet=[sql, bigtable]}>} in []
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:6)
	at org.apache.beam.sdk.metrics.MetricsTest.assertStringSetMetrics(MetricsTest.java:449)
```

is this expected? (there is still some gap to make this metrics available in PipelineResult

**Comment by robertwb** [comment]
> is this expected? (there is still some gap to make this metrics available in PipelineResult

Yes, there's still work to plumb the results back on the Dataflow service end.

File: runners/core-java/src/main/java/org/apache/beam/runners/core/metrics/MetricsContainerImpl.java [hunk 1]
```diff
@@ -391,6 +423,15 @@ public Map<String, ByteString> getMonitoringData(ShortIdMap shortIds) {
             }
           }
         });
+    stringSets.forEach(
+        (metricName, gaugeCell) -> {
```
Comment (Abacn) [inline runners/core-java/src/main/java/org/apache/beam/runners/core/metrics/MetricsContainerImpl.java:427]: minor: gaugeCell -> stringSetSell

**Review by Abacn — APPROVED** [review]
thanks

File: runners/core-java/src/main/java/org/apache/beam/runners/core/metrics/MetricsContainerImpl.java (hunk 1 above)
Comment (robertwb) [reply runners/core-java/src/main/java/org/apache/beam/runners/core/metrics/MetricsContainerImpl.java]: Done.

**Pull request merged**